### PR TITLE
Announcement modal

### PR DESCRIPTION
**Pull request by Zerrien**
This pull request should:

Display an announcement modal to users who return 18 hours later. They should only see the latest announcement modal (as dictated by the server manifest) if their current lastSeen isn't the latest from the server.

There should be a tracker event when the announcement modal is shown, and when the Subscriber button is pressed.

Non-subscribers should see "Subscribe Now" and Subscribers should see "Okay". Non-subscribers should have a link to the premium features page in the info box, while subscribers should have a link to The Second Kithmaze if they've completed 7 levels, otherwise tell them to play a bit more.

Mousing over the pets in the announcement modal animates them, and displays a tooltip with a name and information.

### CODE REVIEW

File: app/schemas/events/show_announcement_modal.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    "version": "1-0-0"
+  },
+  "type": "object",
+  "properties": {
+    "user": {"type": "string", "pattern": "^[a-z0-9]{24}$", "maxLength": 24, "minLength": 24},
+    "label": {"type": "string"}
+  },
+  "required": ["user", "label"],
+  "additionalProperties": false
+}
```
Comment (basicer) [inline app/schemas/events/show_announcement_modal.json]: Your missing an new line at end of file.

File: app/styles/play/modal/announcement-modal.sass [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+    z-index: 1
+    width: 256px
+
+    button
+      width: 256px
+
+  .shimmy
+    animation: shimmy-and-shake 0.25s 2 alternate
+
+  @keyframes shimmy-and-shake
```
Comment (basicer) [inline app/styles/play/modal/announcement-modal.sass]: Is this the best name for this?  It sure is fun though.

File: app/views/play/CampaignView.coffee [hunk 3]
```diff
@@ -980,3 +983,19 @@ module.exports = class CampaignView extends RootView
...
+    latest = window.serverConfig.latestAnnouncement
+    myLatest = me.get('lastAnnouncementSeen')
+    console.log "LID: #{latest} vs #{myLatest}"
+    return unless typeof latest is 'number'
+    accountHours = (new Date() - new Date(me.get("dateCreated"))) / (60 * 60 * 1000) # min*sec*ms
+    return unless accountHours > 18
+    if latest > myLatest or not myLatest?
+      me.set('lastAnnouncementSeen', latest)
+      me.save()
+      console.log "Opening #{latest}"
```
Comment (basicer) [inline app/views/play/CampaignView.coffee]: You have a console log.

File: app/schemas/events/show_announcement_modal.json (hunk 1 above)
Comment (Zerrien) [inline app/schemas/events/show_announcement_modal.json]: Fixed (and cleared up the extra newlines at the end of the AnnouncementModal.coffee)

File: app/styles/play/modal/announcement-modal.sass (hunk 2 above)
Comment (Zerrien) [inline app/styles/play/modal/announcement-modal.sass]: Renamed it to 'excited wiggle' and the class to 'wiggle' so it more descriptive.

A shimmy is more of a slide back and forth, but a wiggle is a bit more of a in-place squirm.

File: app/views/play/CampaignView.coffee (hunk 3 above)
Comment (Zerrien) [inline app/views/play/CampaignView.coffee]: Removed both of them.

**Pull request merged**